feat(js): Add documentation for new `ContextLines` integration (#7564)
getsentry/sentry-docs: src/includes/platforms/configuration/integrations/plugin.mdx

<PlatformContent includePath="configuration/reporting-observer" />

</PlatformSection>

<PlatformSection supported={["javascript"]}>

### ContextLines

_Import name: `Sentry.Integrations.ReportingObserver`_

> This integration collects code lines at and around the stack frames of an error that point to code in the current page's html (for example in `<script>` tags).

Author: much better! thx for the suggestion

<PlatformContent includePath="configuration/reporting-observer" />

</PlatformSection>

<PlatformSection supported={["javascript"]}>

### ContextLines

_Import name: `Sentry.Integrations.ContextLines`_

> This integration attaches source code of inline JavaScript inside the page's HTML (e.g. JS in `<script>` tags) to errors captured by the SDK.

Reviewer: ```suggestion
This integration adds source file context to stack traces for captured errors from the inline JavaScript inside the page's HTML (e.g. JS in `<script>` tags).

```

Author: I tweaked this a little further but thanks for the suggestion!

> Note that this integration _cannot_ collect source code from assets referenced by your HTML (e.g. `<script src="..." />`).

Reviewer: ```suggestion
This integration _can't_ collect source code from assets referenced by your HTML (e.g. `<script src="..." />`).
```

> Generally, the Sentry backend tries to attach source code to these errors by fetching the page and extracting the respective lines.

Reviewer: ```suggestion
This integration is useful when you have inline JS code in HTML pages that can't be accessed by Sentry's backend, for example, due to a login-protected page.

If Sentry can fetch the relevant JavaScript file this integration isn't needed. 
```

Reviewer: I tried to combine the sentences here to be more concise and user-focused. Feel free to tweak if I have any details wrong.

> However, in some cases, our backend might not be able to fetch the JavaScript file, for example if the file is only available for authenticated users.

Reviewer: ```suggestion
```

> Use this integration if you have inline JS code in HTML pages that can't be accessed by our backend, for example due to a login-protected page.

Reviewer: ```suggestion
```

move up
